packages/react-instantsearch-hooks-dom/src/ui/Pagination.tsx

```diff
@@ -0,0 +1,168 @@
+        props.className
+      )}
+    >
+      <ul className="ais-Pagination-list">
+        {showFirst && (
+          <PaginationItem
+            isDisabled={isFirstPage}
+            className="ais-Pagination-item--firstPage"
+            aria-label="First"
+            href={createURL(firstPageIndex)}
+            onClick={() => onNavigate(firstPageIndex)}
+          >
+            ‹‹
+          </PaginationItem>
+        )}
+        {showPrevious && (
+          <PaginationItem
+            isDisabled={isFirstPage}
+            className="ais-Pagination-item--previousPage"
+            aria-label="Previous"
+            href={createURL(previousPageIndex)}
+            onClick={() => onNavigate(previousPageIndex)}
+          >
+            ‹
+          </PaginationItem>
+        )}
+        {pages.map((page) => {
+          const label = String(page + 1);
+
+          return (
+            <PaginationItem
+              key={page}
+              isDisabled={false}
+              className={cx(
+                'ais-Pagination-item--page',
+                page === currentPage && 'ais-Pagination-item--selected'
+              )}
+              aria-label={label}
+              href={createURL(page)}
+              onClick={() => onNavigate(page)}
+            >
+              {label}
+            </PaginationItem>
+          );
+        })}
+        {showNext && (
+          <PaginationItem
+            isDisabled={isLastPage}
+            className="ais-Pagination-item--nextPage"
+            aria-label="Next"
+            href={createURL(nextPageIndex)}
+            onClick={() => onNavigate(nextPageIndex)}
+          >
+            ›
+          </PaginationItem>
+        )}
+        {showLast && (
+          <PaginationItem
+            isDisabled={isLastPage}
+            className="ais-Pagination-item--lastPage"
+            aria-label="Last"
+            href={createURL(lastPageIndex)}
+            onClick={() => onNavigate(lastPageIndex)}
+          >
+            ››
+          </PaginationItem>
+        )}
+      </ul>
+    </div>
+  );
+}
+
+type PaginationItemProps = Omit<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  'onClick'
+> & {
+  onClick: NonNullable<
+    React.AnchorHTMLAttributes<HTMLAnchorElement>['onClick']
+  >;
+} & {
+  isDisabled: boolean;
+};
+
+function PaginationItem({
+  isDisabled,
+  className,
+  href,
+  onClick,
+  ...props
+}: PaginationItemProps) {
+  if (isDisabled) {
+    return (
+      <li
+        className={cx(
+          'ais-Pagination-item ais-Pagination-item--disabled',
+          className
+        )}
+      >
+        <span className="ais-Pagination-link" {...props} />
+      </li>
+    );
+  }
+
+  return (
+    <li className={cx('ais-Pagination-item', className)}>
+      <a
+        className="ais-Pagination-link"
+        href={href}
+        onClick={(event) => {
+          if (isModifierClick(event)) {
+            return;
+          }
+
+          event.preventDefault();
```

Comment:
This prevents from reloading the page. Not sure how to properly test this though.

Reply:
you can check event.defaultPrevented or similar, but not sure if that is worth testing. If the click is not prevented, the js stops right (or does the testing utility not implement that?)

Reply:
If the click isn't prevented, the JS is never executed because the `<a>` behavior takes over (it navigates) so indeed this is implicitly tested by the `expect(props.onNavigate).toHaveBeenCalledTimes(n)` in other tests 👍